### PR TITLE
test: fix test-loopback and test-resolved-stream on Ubuntu 24.04

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/hostname/hostnamectl.c [hunk 1]
```diff
@@ -427,11 +427,11 @@ static int show_all_names(sd_bus *bus) {
                             "b",
                             false);
         if (r < 0) {
-                log_full_errno(sd_bus_error_has_names(
+                log_full_errno((sd_bus_error_has_names(
                                                &error,
                                                BUS_ERROR_NO_PRODUCT_UUID,
                                                SD_BUS_ERROR_INTERACTIVE_AUTHORIZATION_REQUIRED,
-                                               SD_BUS_ERROR_UNKNOWN_METHOD) ? LOG_DEBUG : LOG_WARNING,
+                                               SD_BUS_ERROR_UNKNOWN_METHOD) || r == -ENOENT) ? LOG_DEBUG : LOG_WARNING,
```
Comment (yuwata) [inline src/hostname/hostnamectl.c]: Hmm, why the method does not reply `BUS_ERROR_NO_PRODUCT_UUID` ??

File: src/hostname/hostnamectl.c [hunk 2]
```diff
@@ -448,11 +448,11 @@ static int show_all_names(sd_bus *bus) {
                             &hardware_serial_reply,
                             NULL);
         if (r < 0)
-                log_full_errno(sd_bus_error_has_names(
+                log_full_errno((sd_bus_error_has_names(
                                                &error,
                                                BUS_ERROR_NO_HARDWARE_SERIAL,
                                                SD_BUS_ERROR_INTERACTIVE_AUTHORIZATION_REQUIRED,
-                                               SD_BUS_ERROR_UNKNOWN_METHOD) ? LOG_DEBUG : LOG_WARNING,
+                                               SD_BUS_ERROR_UNKNOWN_METHOD) || r == -ENOENT) ? LOG_DEBUG : LOG_WARNING,
```
Comment (yuwata) [inline src/hostname/hostnamectl.c]: Also, here. Maybe the issue is in the service side?

Comment (bluca) [reply src/hostname/hostnamectl.c]: Haven't got a scooby. I don't see anything obvious, but then again I didn't implement the feature, just trying to get the CI back to green. Let's do that first please, then @poettering can look in more details and change this if there's a better solution

**Comment by poettering** [comment]
I *really* don't grok what's going on here. I don't see how this could possible return ENOENT. Very very confusing.

I wonder if the the hw serial thing is actually a red herring though, and the real problem is something with PK not being around in the test case?

**Comment by bluca** [comment]
I don't think polkit is installed, but it never was in this smoke test. But it shouldn't cause errors to be printed when that's the case.

**Comment by bluca** [comment]
I built an mkosi debian image, removed polkit, hostnamectl doesn't print any errors

File: src/resolve/test-resolved-stream.c [hunk 3]
```diff
@@ -356,20 +356,25 @@ static void try_isolate_network(void) {
...
-        /* Bring up the loopback interfaceon the newly created network namespace */
+        /* Bring up the loopback interface on the newly created network namespace */
         struct ifreq req = { .ifr_ifindex = 1 };
         assert_se((socket_fd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0)) >= 0);
         assert_se(ioctl(socket_fd, SIOCGIFNAME, &req) >= 0);
         assert_se(ioctl(socket_fd, SIOCGIFFLAGS, &req) >= 0);
         assert_se(FLAGS_SET(req.ifr_flags, IFF_LOOPBACK));
         req.ifr_flags |= IFF_UP;
-        assert_se(ioctl(socket_fd, SIOCSIFFLAGS, &req) >= 0);
+        /* Do not assert on this, fails in the Ubuntu Noble CI environment */
```
Comment (poettering) [inline src/resolve/test-resolved-stream.c]: what is Noble CI?

File: src/resolve/test-resolved-stream.c [hunk 4]
```diff
@@ -356,20 +356,25 @@ static void try_isolate_network(void) {
...
+        r = RET_NERRNO(ioctl(socket_fd, SIOCSIFFLAGS, &req));
+        if (r < 0)
+                return -EPERM;
```
Comment (poettering) [inline src/resolve/test-resolved-stream.c]: hardcoded to EPERM?

why not return `r` here?

and why does this fail? seccomp?

Comment (poettering) [reply src/resolve/test-resolved-stream.c]: i mean, why bother with RET_NERRNO() if you ignore the error?

File: src/resolve/test-resolved-stream.c (hunk 3 above)
Comment (bluca) [reply src/resolve/test-resolved-stream.c]: Noble is the codename of Ubuntu 24.04

File: src/resolve/test-resolved-stream.c (hunk 4 above)
Comment (bluca) [reply src/resolve/test-resolved-stream.c]: because I added it after the return

Comment (poettering) [reply src/resolve/test-resolved-stream.c]: hmm? what? i don't follow?

Comment (bluca) [reply src/resolve/test-resolved-stream.c]: It was a mistake - first I added the return -EPERM if ioctl failed, then added the RET_NERRNO, but forgot to update the return, it's fixed now

File: src/hostname/hostnamectl.c [hunk 5]
```diff
@@ -427,11 +427,11 @@ static int show_all_names(sd_bus *bus) {
                             "b",
                             false);
         if (r < 0) {
-                log_full_errno(sd_bus_error_has_names(
+                log_full_errno((sd_bus_error_has_names(
                                                &error,
                                                BUS_ERROR_NO_PRODUCT_UUID,
                                                SD_BUS_ERROR_INTERACTIVE_AUTHORIZATION_REQUIRED,
-                                               SD_BUS_ERROR_UNKNOWN_METHOD) ? LOG_DEBUG : LOG_WARNING,
```
Comment (YHNdnzj) [inline src/hostname/hostnamectl.c]: As mentioned, it's not necessary to do this for GetProductUUID. The method was introduced along with BUS_ERROR_NO_PRODUCT_UUID.

File: src/resolve/test-resolved-stream.c [hunk 6]
```diff
@@ -356,20 +356,25 @@ static void try_isolate_network(void) {
...
+        /* Bring up the loopback interface on the newly created network namespace */
         struct ifreq req = { .ifr_ifindex = 1 };
         assert_se((socket_fd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0)) >= 0);
         assert_se(ioctl(socket_fd, SIOCGIFNAME, &req) >= 0);
         assert_se(ioctl(socket_fd, SIOCGIFFLAGS, &req) >= 0);
         assert_se(FLAGS_SET(req.ifr_flags, IFF_LOOPBACK));
         req.ifr_flags |= IFF_UP;
-        assert_se(ioctl(socket_fd, SIOCSIFFLAGS, &req) >= 0);
+        /* Do not assert on this, fails in the Ubuntu 24.04 CI environment */
+        r = RET_NERRNO(ioctl(socket_fd, SIOCSIFFLAGS, &req));
```
Comment (poettering) [inline src/resolve/test-resolved-stream.c:373]: mabye add here:

```c
assert(ERRNO_IS_NEG_PRIVILEGE(r));
```

i.e. just to catch unexpected errors here, i.e. any that are not about privs

File: src/resolve/test-resolved-stream.c [hunk 7]
```diff
@@ -381,7 +386,8 @@ int main(int argc, char **argv) {
 
         test_setup_logging(LOG_DEBUG);
 
-        try_isolate_network();
+        if (try_isolate_network() < 0)
+                return log_tests_skipped("lacking privileges");
```
Comment (poettering) [inline src/resolve/test-resolved-stream.c:392]: or alternatively do this here:

```c
r = try_isolate_network();
if (ERRNO_IS_NEG_PRIVILEGE(r)) 
        return log_tests_skipped("lacking privileges");
assert_se(r >= 0);
```

File: src/resolve/test-resolved-stream.c (hunk 6 above)
Comment (poettering) [reply src/resolve/test-resolved-stream.c:373]: i mean the point of test cases is to catch unexpected errors after all, hence i think there's value in catching the precise errors

File: src/resolve/test-resolved-stream.c (hunk 7 above)
Comment (poettering) [reply src/resolve/test-resolved-stream.c:392]: i much prefer this solution btw

**Comment by poettering** [comment]
lgtm

**Pull request merged**